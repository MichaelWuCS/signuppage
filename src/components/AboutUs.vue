<template>
  <div>
    {{ message }} <br />
    <PokemonSelector> </PokemonSelector>
    {{ pokemon }}
  </div>
</template>

<script>
import Axios from 'axios';
import PokemonSelector from '@/components/PokemonSelector.vue';
export default {
  components: {
    PokemonSelector,
  },
  props: {
    message: String,
  },
  data() {
    return {
      pokemon: null,
    };
  },
  async created() {
    const response = await Axios.get('https://pokeapi.co/api/v2/pokemon/ditto');
    this.pokemon = response.data;
  },
};
</script>
