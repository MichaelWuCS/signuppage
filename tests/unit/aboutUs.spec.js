import AboutUs from '@/components/AboutUs.vue';
import { shallowMount, mount } from '@vue/test-utils';
jest.mock('axios', () => {
  return { get: jest.fn() };
});
import Axios from 'axios';

describe('aboutUs', () => {
  test('aboutUs should render the selected pokemon information (ditto)', async () => {
    //Arrange
    const messageProp = 'Hello';
    Axios.get.mockImplementation(async () => {
      return {
        data: 'mockDittoData',
      };
    });

    //Act
    const wrapper = shallowMount(AboutUs, {
      propsData: { message: messageProp },
    });
    //await new Promise((resolve) => setTimeout(resolve, 1000));
    await wrapper.vm.$nextTick();
    const result = wrapper.html();
    //Assert
    expect(result).toMatchSnapshot();
  });
});
