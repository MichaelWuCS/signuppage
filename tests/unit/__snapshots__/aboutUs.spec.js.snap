// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aboutUs aboutUs should render the selected pokemon information (ditto) 1`] = `
<div>
  Hello <br>
  <pokemonselector-stub></pokemonselector-stub>
  mockDittoData
</div>
`;
